src/main/java/com/coveo/spillway/storage/InMemoryStorage.java

```diff
@@ -57,7 +58,7 @@
 
   @Override
   public Map<LimitKey, Integer> addAndGet(Collection<AddAndGetRequest> requests) {
-    Map<LimitKey, Integer> updatedEntries = new HashMap<>();
+    Map<LimitKey, Integer> updatedEntries = new LinkedHashMap<>();
     synchronized (lock) {
       for (AddAndGetRequest request : requests) {
```

Comment:
the InMemoryStorage isnt used in the sliding window so I prefer to let it for another day haha.
I just changed the map to a LinkedHashMap because I realized that we were doing some logic based on sequential reading while using a regular HashMap... @GuiSim was it intentional?